spec/controllers/application_templates_controller_spec.rb

```diff
@@ -173,4 +173,54 @@
       end
     end
   end
+
+  describe 'POST #toggle_unavailability_enabled' do
+    before :each do
+      department = create :department, name: 'Bus'
+      position = create :position, department: department, name: 'Operator'
+      @template = create :application_template, position: position
+    end
+    let :submit do
+      post :toggle_unavailability_enabled,
+           id: @template.id,
+           position: @template.position.name,
+           department: @template.department.name
+    end
+    context 'staff' do
+      before :each do
+        when_current_user_is :staff
+        request.env['HTTP_REFERER'] = 'http://test.host/redirect'
+      end
+      it 'changes unavailability_enabled when called' do
+        expect { submit }.to change { @template.reload.unavailability_enabled }
+      end
+
+      context 'unavailability is disabled' do
+        before :each do
+          # First, disable EEO
+          @template.update unavailability_enabled: false
+        end
+        it 'enables unavailability' do
+          submit
+          expect(@template.reload.unavailability_enabled).to be true
+        end
+      end
+
+      context 'unavailability is enabled' do
+        before :each do
+          # First, enable EEO
+          @template.update unavailability_enabled: true
+        end
+        it 'disbles unavailability' do
+          submit
+          expect(@template.reload.unavailability_enabled).to be false
+        end
+      end
+
+      it 'redirects back' do
+        submit
```

Comment:
Good news: because [we have a custom matcher for this](https://github.com/umts/jobapps/commit/49ef81c238b4909ed8fb58da9acd949010a0ef36#diff-8b7db4d5cc4b8f6dc8feb7030baa2478R48), you can just write

```ruby
expect { submit }.to redirect_back
```

... and you can leave out setting the `HTTP_REFERER` in your test setup.